### ISSUE TITLE
ShowMySky: Color "jump" at begin/end of eclipse time

### ISSUE THREAD

**Issue by gzotti** [Stellarium/stellarium#2914]
ShowMySky provides a magnificent "eclipse sky" simulation. It is graphically much more involved (i.e. shows only a few fps during eclipse) than regular daylight sky, so it is only used during at least minimal coverage of the sun by the moon. 

### Expected Behaviour

Transition between regular and eclipse mode goes unnoticed in the sky color.

### Actual Behaviour

In all eclipse mode settings:
At the time of first (or last) contact, the blue shade of the sky changes. 

### Steps to reproduce

Check with any Solar eclipse. 

I don't know the details of this issue, do we just have different atmosphere compositions, or is there a systematic flaw?

### System
* Stellarium version: (master on 2022-12-14)
* Operating system: Win11
* Graphics Card: Geforce M 3070Ti 

### Logfile 
If possible, attach the logfile `log.txt` from your user data directory. Look into the Guide for its location.

**Comment by 10110111** [comment]
Indeed, looks bad. I wonder how I didn't notice it before.

> I don't know the details of this issue

The issue is that in eclipsed atmosphere only two scattering orders are considered, while in the common case more orders are included.

**Comment by Atque** [comment]
Is this bug still present? I can't reproduce it.

**Comment by luzpaz** [comment]
@Atque can you specify the version you tested ?

**Comment by Atque** [comment]
@luzpaz Okay, I can reproduce this with eclipse quality 2 and 3, but not with 1.

**Comment by alex-w** [comment]
@10110111 Is this bug still present?

**Comment by 10110111** [comment]
Yes, you can check this by going to the last contact observation position of the eclipse of 2026-08-12, change time to 20:05:40 UTC, wait until 20:05:42 and notice that the colors change a bit.
